databroker/src/grpc/kuksa_val_v1/val.rs

```diff
@@ -143,17 +148,56 @@ impl proto::val_server::Val for broker::DataBroker {
                                     }
                                 }
                             }
-                        }
-
-                        // If there are result fields, add them to the entries list.
-                        if !result_fields.is_empty() {
-                            let proto_entry =
-                                proto_entry_from_entry_and_fields(entry, result_fields);
-                            debug!("Getting datapoint: {:?}", proto_entry);
-                            entries.push(proto_entry);
-                        }
-                    })
-                    .await;
+                            // If there are result fields, add them to the entries list.
+                            if !result_fields.is_empty() {
+                                let proto_entry =
+                                    proto_entry_from_entry_and_fields(entry, result_fields);
+                                debug!("Getting datapoint: {:?}", proto_entry);
+                                entries.push(proto_entry);
+                            }
+                        })
+                        .await;
+
+                    // Not found any matches meaning it could be a branch path request
+                    if matcher.is_branch_matcher() && !(*is_match) {
+                        //*glob += "/**";
+                        broker
+                            .for_each_entry(|entry| {
+                                let mut result_fields: HashSet<proto::Field> = HashSet::new();
+                                let glob_path = &entry.metadata().glob_path;
+                                if matcher.glob_match(glob_path) {
```

Comment:
Isn't this:
```rust
if matcher.glob_match(glob_path) {
```

doing the exact same thing... again?

And why do this only if the first iteration didn't match. We are doing a glob match anyway. Can we not match both branches and not branches with a single pattern at the same time?